assets/src/components/AssignmentTable.js

```diff
@@ -120,170 +144,272 @@ function AssignmentTable (props) {
     }
   }, [currentWeekRow.current])
 
+  const [assignmentGroupNames, setAssignmentGroupNames] = useState([])
+  useEffect(() => {
+    const allGroupNames = assignments.map(ag => ag.assignmentGroup.name)
+    var uniqueGroupNames = [...new Set(allGroupNames)]
+    uniqueGroupNames.sort()
+    setAssignmentGroupNames(uniqueGroupNames)
+  }, [assignments])
+
+  const [assignmentFilter, setAssignmentFilter] = useState('')
+  function handleNameFilterChange (e) {
+    setAssignmentFilter(e.target.value)
+  }
+
+  const ITEM_HEIGHT = 48
+  const ITEM_PADDING_TOP = 8
+  const MenuProps = {
+    PaperProps: {
+      style: {
+        maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+        width: 250
+      }
+    }
+  }
+
+  const [assignmentGroupFilterArray, setAssignmentGroupFilterArray] = useState([])
+  function handleGroupNameArrayFilterChange (e) {
+    setAssignmentGroupFilterArray(e.target.value)
+  }
+
+  function handleGradedOnlyFilterChange (e) {
+    setGradedOnly(!gradedOnly)
+  }
+
+  const TRUNCATED_GROUP_NAME_LENGTH = 30
+  const shortenString = text => {
+    return (text.length > TRUNCATED_GROUP_NAME_LENGTH)
+      ? text.substring(0, TRUNCATED_GROUP_NAME_LENGTH - 3) + '...'
+      : text
+  }
+
   return (
     <RootRef rootRef={tableRef}>
-      <TableContainer className={classes.container}>
-        <MTable stickyHeader ref={tableRef}>
-          <TableHead>
-            <TableRow>
-              {
-                [
-                  'Week',
-                  'Due',
-                  'Assignment Name',
-                  'Percent of Final Grade',
-                  'Score / Out of',
-                  'Lock Goal'
-                ].map((heading, key) => (
-                  <TableCell
-                    className={classes.tableCell + ' ' + classes.tableHeadCell}
-                    key={key}
-                  >
-                    {heading}
-                  </TableCell>
-                ))
+      <div>
+        <Grid container className={classes.filterArea}>
+          {/* Hidden until UI requirements are hashed out */}
+          <Grid item xs={12} sm={5}>
+            <FormControl className={classes.formControl}>
+              <InputLabel>Assignment Group</InputLabel>
```

Comment:
This Form is outside of the table, and we might be run into visibility issue as screen size varies. Since stacking order of element might not work perfectly. On a regular size laptop should work ok. for #1113 I am trying to minimize the top screen real estate I may need to consider this layout as well

Reply:
It is outside the table (the table doesn't support filtering..) but it is [responsive ](https://material-ui.com/components/grid/)so the filter itself should respond ok to various screen sizes.